gammapy/irf/background.py

```diff
@@ -114,3 +115,17 @@ def read(cls, filename, hdu='BACKGROUND'):
         filename = make_path(filename)
         hdulist = fits.open(str(filename))
         return cls.from_hdulist(hdulist, hdu=hdu)
+
+    def to_table(self):
+        #TODO: BGD or BKG - see https://gamma-astro-data-formats.readthedocs.io/en/latest/irfs/full_enclosure/bkg/index.html
+        table = Table()
+        table['DETX_LO'] = [self.data.axis('detx').hi]*self.data.axis('detx').unit
+        table['DETX_HI'] = [self.data.axis('detx').hi]*self.data.axis('detx').unit
+        table['DETY_LO'] = [self.data.axis('dety').lo]*self.data.axis('dety').unit
+        table['DETY_HI'] = [self.data.axis('dety').hi]*self.data.axis('dety').unit
+        table['ENERG_LO'] = [self.data.axis('energy').lo]*self.data.axis('energy').unit
+        table['ENERG_HI'] = [self.data.axis('energy').hi]*self.data.axis('energy').unit
+        table['BGD'] = [self.data.data]*self.data.data.unit
+        self.meta.update({'name':'BACKGROUND'})
```

Comment:
I think it would be best if the "to_table" method didn't modify `self`. It's supposed to be a serialisation, so the `self` object should remain unchanged IMO.

Would this work?
```
meta = self.meta.copy()
table = Table(meta=meta)
```
Is adding the `name` key really needed or useful? If yes, you could add it to this copy of the `meta` dict here, or move it to some other place, e.g. `__init__`.

This comment applies to all `to_table` methods.

Reply:
It could be taken out all together, but the user would have to make sure to include it when adding the provenance data. The function table_to_fits_table extracts this in order to assign the name to the fits table. An alternative/additional solution could be:

```python
def table_to_fits_table(table, name=None):
    ...
    if name is None:
         name = table.meta.pop('name', None)
```
which could allow this to be placed earlier (i.e. to_fits could have the same format). Either way, should I still include the 
```python
meta = self.meta.copy()
table = Table(meta=meta)
```
so that when additional information is included (such as in the example) it doesn't modify self?

Reply:
Hmm ... OK, now I see that handling of `name` is difficult because:

* [BinTableHDU](http://docs.astropy.org/en/stable/io/fits/api/tables.html#astropy.io.fits.BinTableHDU) has `name` separate from `header` (but presumably they have some auto-interaction e.g. on read / write with header keywords).
* [Table](http://docs.astropy.org/en/stable/api/astropy.table.Table.html) doesn't have a `name` attribute, so the urge to combine `header` and `name` into meta is strong, but then the question is what all the places are where to do this.

I see that in `Table.read` and `Table.write` the HDU name round-trips through the `EXTNAME` key in `meta`:
```
In [1]: from astropy.table import Table

In [2]: t = Table.read('http://gamma-astro-data-formats.readthedocs.io/en/latest/_downloads/aeff_2d_full_example.fits')
Downloading http://gamma-astro-data-formats.readthedocs.io/en/latest/_downloads/aeff_2d_full_example.fits
|=======================================================================================================================|  14k/ 14k (100.00%)         0s
WARNING: hdu= was not specified but multiple tables are present, reading in first available table (hdu=1) [astropy.io.fits.connect]

In [3]: t.meta
Out[3]: 
OrderedDict([('OBS_ID', 31415),
             ('LO_THRES', 0.1),
             ('HI_THRES', 50),
             ('HDUDOC',
              'https://github.com/open-gamma-ray-astro/gamma-astro-data-formats'),
             ('HDUVERS', '0.2'),
             ('HDUCLASS', 'GADF'),
             ('HDUCLAS1', 'RESPONSE'),
             ('HDUCLAS2', 'EFF_AREA'),
             ('HDUCLAS3', 'FULL-ENCLOSURE'),
             ('HDUCLAS4', 'AEFF_2D'),
             ('EXTNAME', 'EFFECTIVE AREA')])

In [4]: t.write('/tmp/test.fits')
WARNING: UnitsWarning: 'None' did not parse as fits unit: At col 0, Unit 'None' not supported by the FITS standard.  [astropy.units.core]

In [5]: from astropy.io import fits

In [6]: hdu_list = fits.open('/tmp/test.fits')

In [7]: hdu_list.info()
Filename: /tmp/test.fits
No.    Name      Ver    Type      Cards   Dimensions   Format
  0  PRIMARY       1 PrimaryHDU       4   ()      
  1  EFFECTIVE AREA    1 BinTableHDU     35   1R x 5C   [20D, 20D, 5D, 5D, 100D]   
```
But as far as I know they have no support for writing multiple HDUs to file, this would clobber, but by default fails if the file exists:
```
t.meta['EXTNAME'] = 'AEFF V2'   
t.write('/tmp/test.fits') 
```

So basically I think sane FITS extension name handling is an unsolved issue in Gammapy (and probably ctapipe)?
@thomasarmstrong - I think maybe what you suggest could be the best way to proceed: add a `name` option for the functions and methods that convert to `BinTableHDU`. But probably it should default to the key `EXTNAME` in `meta`, not `name` like it is now?

It's a bit perverse, but this is the hardest part of this PR. So if you can add tests that actually write to FITS files to establish the current behaviour and show that it works, i.e. a caller can have control over HDU names that get written, that would be useful.